tests/Platforms/AbstractPlatformTestCase.php

```diff
@@ -1218,6 +1218,37 @@ public static function getEnumDeclarationSQLProvider(): array
         ];
     }
 
+    /** @param array<string> $values */
+    #[DataProvider('getEnumDeclarationWithLengthSQLProvider')]
+    public function testGetEnumDeclarationWithLengthSQL(array $values, int $length, ?string $expectedSQL = null): void
+    {
+        if ($expectedSQL === null) {
+            $this->expectException(InvalidArgumentException::class);
+        }
```

Comment:
This test covers the happy case as well as the exceptional one. I find this confusing and in fact I overlooked the exceptional part at first. Can you please split this into two tests so we don't need those two `if ($expectedSQL === null)` blocks?